nominal/_checklist.py

```diff
@@ -0,0 +1,466 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING, Literal, Mapping, Sequence
+
+from typing_extensions import Self
+
+from ._api.combined import (
+    api,
+    scout_api,
+    scout_checks_api,
+    scout_compute_api,
+    scout_compute_representation_api,
+    scout_run_api,
+)
+
+if TYPE_CHECKING:
+    # prevent circular imports
+    from .core import NominalClient
+
+
+# TODO(ritwikdixit): add support for more fields i.e. lineage
+@dataclass(frozen=True)
+class Check:
+    rid: str
+    name: str
+    expression: str
+    priority: Priority
+    description: str
+    _client: NominalClient = field(repr=False)
+
+
+@dataclass(frozen=True)
+class ChecklistVariable:
+    name: str
+    expression: str
+    _client: NominalClient = field(repr=False)
+
+
+@dataclass(frozen=True)
+class ChecklistBuilder:
+    name: str
+    assignee_email: str
+    description: str
+    _default_ref_name: str | None
+    # note that the ChecklistBuilder is immutable, but the lists/dicts it contains are mutable
+    _variables: list[_CreateChecklistVariable]
+    _checks: list[_CreateCheck]
+    _properties: dict[str, str]
+    _labels: list[str]
```

Comment:
we actually mutate these so we don't use the generic `Sequence`/`Mapping` types. We could use `MutableSequence`/`MutableMapping` but since these are internal fields we can change the types